File: solutions/day01/solution.go
package main

import "fmt"

func numWaterBottles(numBottles int, numExchange int) int {
	total := numBottles
	empty := numBottles

	for empty >= numExchange {
		newBottles := empty / numExchange
		total += newBottles
		empty = empty%numExchange + newBottles
	}

	return total
}

func main() {
	numBottles := 9
	numExchange := 3
	fmt.Println(numWaterBottles(numBottles, numExchange)) // Output: 13
}

File: solutions/day01/solution.js
var numWaterBottles = function(numBottles, numExchange) {
    let totalDrank = numBottles;
    let emptyBottles = numBottles;

    while (emptyBottles >= numExchange) {
        let newBottles = Math.floor(emptyBottles / numExchange);
        totalDrank += newBottles;
        emptyBottles = newBottles + (emptyBottles % numExchange);
    }

    return totalDrank;
};
